cluster/gce/gci/nvidia-gpus/installer.sh

```diff
@@ -0,0 +1,202 @@
+# ${BASE_DIR}/.cache/* --> Nvidia driver artifacts cached for idempotency.
+#
+
+BASE_DIR=${BASE_DIR:-"/rootfs/nvidia"}
+CACHE_DIR="${BASE_DIR}/.cache"
+USR_WORK_DIR="${CACHE_DIR}/usr-work"
+USR_WRITABLE_DIR="${CACHE_DIR}/usr-writable"
+LIB_WORK_DIR="${CACHE_DIR}/lib-work"
+LIB_WRITABLE_DIR="${CACHE_DIR}/lib-writable"
+
+LIB_OUTPUT_DIR="${BASE_DIR}/lib"
+BIN_OUTPUT_DIR="${BASE_DIR}/bin"
+
+KERNEL_SRC_DIR="/lakitu-kernel"
+NVIDIA_DRIVER_DIR="/nvidia"
+NVIDIA_DRIVER_VERSION="375.26"
+
+# Source: https://developer.nvidia.com/cuda-downloads
+NVIDIA_CUDA_URL="https://developer.nvidia.com/compute/cuda/8.0/Prod2/local_installers/cuda_8.0.61_375.26_linux-run"
+NVIDIA_CUDA_MD5SUM="33e1bd980e91af4e55f3ef835c103f9b"
+NVIDIA_CUDA_PKG_NAME="cuda_8.0.61_375.26_linux.run"
+NVIDIA_DRIVER_PKG_NAME="NVIDIA-Linux-x86_64-375.26.run"
+
+check_nvidia_device() {
+    lspci
+    if ! lspci | grep -i -q NVIDIA; then
+        echo "No NVIDIA devices attached to this instance."
+        exit 0
+    fi
+    echo "Found NVIDIA device on this instance."
+}
+
+prepare_kernel_source() {
+    local kernel_git_repo="https://chromium.googlesource.com/chromiumos/third_party/kernel"
+    local kernel_version="$(uname -r)"
+    local kernel_version_stripped="$(echo ${kernel_version} | sed 's/\+//')"
+
+    # Checkout the correct tag.
+    echo "Downloading kernel source at tag ${kernel_version_stripped} ..."
+    pushd "${KERNEL_SRC_DIR}"
+    # TODO: Consume KERNEL SHA1 from COS image directly.
+    # git checkout "tags/v${kernel_version_stripped}"
+    git checkout ${LAKITU_KERNEL_SHA1}
+
+    # Prepare kernel configu and source for modules.
+    echo "Preparing kernel sources ..."
+    zcat "/proc/config.gz" > ".config"
+    make olddefconfig
+    make modules_prepare
+    # Done.
+    popd
+}
+
+download_install_nvidia() {
+    local pkg_name="${NVIDIA_CUDA_PKG_NAME}"
+    local url="${NVIDIA_CUDA_URL}"
+    local log_file_name="${NVIDIA_DRIVER_DIR}/nvidia-installer.log"
+
+    mkdir -p "${NVIDIA_DRIVER_DIR}"
+    pushd "${NVIDIA_DRIVER_DIR}"
+
+    echo "Downloading Nvidia CUDA package from ${url} ..."
+    curl -L -s "${url}" -o "${pkg_name}"
+    echo "${NVIDIA_CUDA_MD5SUM} ${pkg_name}" | md5sum --check
+
+    echo "Extracting Nvidia CUDA package ..."
+    sh ${pkg_name} --extract="$(pwd)"
+
+    echo "Running the Nvidia driver installer ..."
+    if ! sh "${NVIDIA_DRIVER_PKG_NAME}" --kernel-source-path="${KERNEL_SRC_DIR}" --silent --accept-license --keep --log-file-name="${log_file_name}"; then
+        echo "Nvidia installer failed, log below:"
+        echo "==================================="
+        tail -50 "${log_file_name}"
+        echo "==================================="
+        exit 1
+    fi
+    # Create unified memory device file.
+    nvidia-modprobe -c0 -u
+    popd
+}
+
+unlock_loadpin_and_reboot_if_needed() {
+    kernel_cmdline="$(cat /proc/cmdline)"
+    if echo "${kernel_cmdline}" | grep -q "root=/dev/dm-0"; then
```

Comment:
I forgot to point this out in my previous review: please replace the `grep` command by the following.  The current command should work but the following one makes it more robust and consistent:
`grep -q -v "lsm.module_locking=0"`